corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -174,15 +174,18 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
             this.model = this.options.model;
             this.hasError = false;
 
-            var allStickyValues = hqImport("cloudcare/js/formplayer/utils/util").getStickyQueryInputs(),
+            var value = this.model.get('value'),
+                allStickyValues = hqImport("cloudcare/js/formplayer/utils/util").getStickyQueryInputs(),
                 stickyValue = allStickyValues[this.model.get('id')],
-                [searchForBlank, value] = decodeValue(this.model, stickyValue);
-            if (value && !this.model.get('value')) {
-                // Set the value and blank search checkbox from the sticky
-                // values if available and no default is present
-                this.model.set('value', value);
-            }
+                [searchForBlank, stickyValue] = decodeValue(this.model, stickyValue);
             this.model.set('searchForBlank', searchForBlank);
+            if (stickyValue && !value) {  // Sticky values don't override default values
```

Comment:
`value` will never be an empty array, because if the user didn't provide a default value for their multiselect, `decodeValue` will specifically return `undefined` - is that correct?

Reply:
`decodeValue` actually isn't called on `value` here.  This if statement is unchanged from the previous version, save for variable name changes.

It does seem more appropriate to call `decodeValue` on value than special-case `select` values as I've done here.  I swear I considered and rejected that yesterday, but I can't for the life of me imagine why.  That'll probably break date ranges, but let me give it a go anyways.

Reply:
oh, but to answer your question directly, `value` will be `null` if the user didn't provide a default

Reply:
Okay, I'm inclined to keep it like this.  We require the user to input multiselect default values using a format _kinda_ like the one used elsewhere (with `#,#` as a separator), except that the formplayer version also uses that with the empty string to encode "search for blank".  Date range questions use a different format for default values, and that would definitely break here (though it looks like it's already broken for a different reason)

Reply:
cool, thank you for explaining